src/backend/distributed/executor/tuple_destination.c

```diff
@@ -86,7 +93,28 @@ TupleStoreTupleDestPutTuple(TupleDestination *self, Task *task,
 							HeapTuple heapTuple, uint64 tupleLibpqSize)
 {
 	TupleStoreTupleDestination *tupleDest = (TupleStoreTupleDestination *) self;
+
+	/*
+	 * Remote execution sets tupleLibpqSize, however it is 0 for local execution. We prefer
+	 * to use tupleLibpqSize for  the remote execution because that reflects the exact data
+	 * transfer size over the network. For local execution, we rely on the size of the
+	 * tuple.
+	 */
+	uint64 dataSize =
+		tupleLibpqSize > 0 ? tupleLibpqSize : HeapTupleHeaderGetDatumLength(heapTuple);
+	TupleDestinationStats *tupleDestinationStats = self->tupleDestinationStats;
+	if (SubPlanLevel > 0 && tupleDestinationStats != NULL)
```

Comment:
this seems like a more appropriate place, and I like the clean up in the executor

Reply:
It is per subplan

Reply:
Ok, so added one more test and should be ready for review. thanks!